From 5dbdfc727f9e997f8c1ca9f01d5d05075fc5ce93 Mon Sep 17 00:00:00 2001
From: Nuno Santos <nsantos@apache.org>
Date: Tue, 29 Nov 2011 21:41:03 +0000
Subject: [PATCH 1/2] Bug 758853, QPID-3533: qpid-config does not properly set
 some options if value evaluates to False -- applied
 patch from bpdeman

git-svn-id: https://svn.apache.org/repos/asf/qpid/trunk@1208094 13f79535-47bb-0310-9956-ffa450edef68
---
 qpid/tools/src/py/qpid-config |   30 +++++++++++++++---------------
 1 files changed, 15 insertions(+), 15 deletions(-)

diff --git a/qpid/tools/src/py/qpid-config b/qpid/tools/src/py/qpid-config
index bb49b9d..45cf01e 100755
--- a/qpid/tools/src/py/qpid-config
+++ b/qpid/tools/src/py/qpid-config
@@ -228,7 +228,7 @@ def OptionsAndArguments(argv):
         config._recursive = True
     if opts.broker_addr:
         config._host = opts.broker_addr
-    if opts.timeout:
+    if opts.timeout is not None:
         config._connTimeout = opts.timeout
         if config._connTimeout == 0:
             config._connTimeout = None
@@ -244,13 +244,13 @@ def OptionsAndArguments(argv):
         config._clusterDurable = True
     if opts.file:
         config._file = opts.file
-    if opts.file_count:
+    if opts.file_count is not None:
         config._fileCount = opts.file_count
-    if opts.file_size:
+    if opts.file_size is not None:
         config._fileSize = opts.file_size
-    if opts.max_queue_size != None:
+    if opts.max_queue_size is not None:
         config._maxQueueSize = opts.max_queue_size
-    if opts.max_queue_count:
+    if opts.max_queue_count is not None:
         config._maxQueueCount = opts.max_queue_count
     if opts.limit_policy:
            config._limitPolicy = opts.limit_policy
@@ -271,13 +271,13 @@ def OptionsAndArguments(argv):
         config._if_unused = False
     if opts.sasl_mechanism:
         config._sasl_mechanism = opts.sasl_mechanism
-    if opts.flow_stop_size:
+    if opts.flow_stop_size is not None:
         config._flowStopSize = opts.flow_stop_size
-    if opts.flow_resume_size:
+    if opts.flow_resume_size is not None:
         config._flowResumeSize = opts.flow_resume_size
-    if opts.flow_stop_count:
+    if opts.flow_stop_count is not None:
         config._flowStopCount = opts.flow_stop_count
-    if opts.flow_resume_count:
+    if opts.flow_resume_count is not None:
         config._flowResumeCount = opts.flow_resume_count
     if opts.group_header:
         config._msgGroupHeader = opts.group_header
@@ -522,9 +522,9 @@ class BrokerManager:
             declArgs[FILECOUNT] = config._fileCount
             declArgs[FILESIZE]  = config._fileSize
 
-        if config._maxQueueSize != None:
+        if config._maxQueueSize is not None:
             declArgs[MAX_QUEUE_SIZE]  = config._maxQueueSize
-        if config._maxQueueCount:
+        if config._maxQueueCount is not None:
             declArgs[MAX_QUEUE_COUNT]  = config._maxQueueCount
         if config._limitPolicy:
             if config._limitPolicy == "none":
@@ -550,13 +550,13 @@ class BrokerManager:
         if config._eventGeneration:
             declArgs[QUEUE_EVENT_GENERATION]  = config._eventGeneration
 
-        if config._flowStopSize:
+        if config._flowStopSize is not None:
             declArgs[FLOW_STOP_SIZE]  = config._flowStopSize
-        if config._flowResumeSize:
+        if config._flowResumeSize is not None:
             declArgs[FLOW_RESUME_SIZE]  = config._flowResumeSize
-        if config._flowStopCount:
+        if config._flowStopCount is not None:
             declArgs[FLOW_STOP_COUNT]  = config._flowStopCount
-        if config._flowResumeCount:
+        if config._flowResumeCount is not None:
             declArgs[FLOW_RESUME_COUNT]  = config._flowResumeCount
 
         if config._msgGroupHeader:
-- 
1.7.7.6

From 65144d9e0501807bb80015df273a68dd759e88c9 Mon Sep 17 00:00:00 2001
From: Ted Ross <tross@apache.org>
Date: Thu, 5 Apr 2012 18:33:26 +0000
Subject: [PATCH 2/2] Bug 806888, QPID-3938: Updated to track an API change in
 the wrapped Python client API.

git-svn-id: https://svn.apache.org/repos/asf/qpid/trunk@1309998 13f79535-47bb-0310-9956-ffa450edef68
---
 qpid/tools/src/py/qmf-tool |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/qpid/tools/src/py/qmf-tool b/qpid/tools/src/py/qmf-tool
index 8413ca2..db51c96 100755
--- a/qpid/tools/src/py/qmf-tool
+++ b/qpid/tools/src/py/qmf-tool
@@ -266,7 +266,7 @@ class QmfData:
     self.conn_options = conn_options
     self.qmf_options = qmf_options
     self.agent_filter = '[]'
-    self.connection = cqpid.Connection(self.url, self.conn_options)
+    self.connection = cqpid.Connection(self.url, **self.conn_options)
     self.connection.open()
     self.session = qmf2.ConsoleSession(self.connection, self.qmf_options)
     self.session.setAgentFilter(self.agent_filter)
-- 
1.7.7.6

